mlx/array.cpp

```diff
@@ -39,14 +39,31 @@ array::array(const std::complex<float>& val, Dtype dtype /* = complex64 */)
 array::array(
     const std::vector<int>& shape,
     Dtype dtype,
-    std::unique_ptr<Primitive> primitive,
+    std::shared_ptr<Primitive> primitive,
     const std::vector<array>& inputs)
     : array_desc_(std::make_shared<ArrayDesc>(
           shape,
           dtype,
           std::move(primitive),
           inputs)) {}
 
+std::vector<array> array::make_arrays(
```

Comment:
This is the key function to make a set of arrays for multi output primitives. 

1. Constructs the arrays.
2. Records their siblings and their position in the output list